### DISCUSSION TITLE
How to obtain raw result from QueryBuilder?

### DISCUSSION THREAD

**Discussion by ciprian-vsp** [mikro-orm/mikro-orm#4000] [Q&A] [answered]
Hello. Is there a way to obtain the raw result using a query builder?
For example, I have the following query builder and I want to get the raw result
```
const qb = await em.createQueryBuilder(User, 'u')
    .select(['id', 'name', `DATE_PART('day', now() - u.created_at) as days'`])
    .where({ // some where clause here with fields from the entity })
    .orderBy({ //some order by here with fields from the entity })
    .limit(1, 0);
```
Right now, I am using knex to get the raw result, like this: 
```
const result = await em.getConnection().execute(qb.getKnexQuery());
console.log(result[0]); // { days: 11 }
```
It works for me, but feels a bit hackish.
I saw that getResult, getSingleResult, getResultList, and execute return the entity.
I am wondering if there is a better way to do this, something like `qb.getRawResult()`?
Thank you!

**Comment by B4nan** [comment, marked as answer]
You can use `qb.execute('all', false)`, I guess having a better named alias like `getRawResult` makes sense.

**Reply by ciprian-vsp** [reply]
Thanks for the fast response. I forgot to mention that I am using TypeScript.
I tried with `qb.execute('all', false)` but it returns a list of entities (the ones from createQueryBuilder) and I know that underneath, the result is not mapped, and I would have to do some casting like `(result[0] as any)`, but I thought maybe I missed something from the docs on getting the raw result and there is a better / cleaner way to do it.
An alias for `qb.execute('all', false)` like `getRawResult` does sound good, maybe in the future :)

**Reply by B4nan** [reply]
The information about how a property name is mapped to a column name is not present anywhere at the type level. If you keep the second param as `true`, it still won't return entity instances, but it will map the column names to property names, so in a way, the return type will be correct (TS types don't describe instances, only their shape, TS is structurally typed).
